plasmapy/atomic/classes.py

```diff
@@ -0,0 +1,250 @@
+
+            self._atomic_number = _Elements[element]['atomic_number']
+            self._element_name = _Elements[element]['name']
+
+            if element and not isotope and not ion:
+                self._standard_atomic_weight = \
+                    _Elements[element]['atomic_mass'].to(u.kg)
+                self._mass = self._standard_atomic_weight
+            elif element and isotope and not ion:
+                self._isotope_mass = _Isotopes[isotope]['atomic_mass']
+
+            self._is_antimatter = False
+
+        if self._integer_charge is not None:
+            self._electric_charge = self._integer_charge * const.e.si
+        else:
+            self._electric_charge = None
+
+    @property
+    def particle(self):
+        return self._particle_symbol
+
+    @property
+    def element(self):
+        return self._atomic_symbol
+
+    @property
+    def isotope(self):
+        return self._isotope_symbol
+
+    @property
+    def ion(self):
+        return self._ion_symbol
+
+    @property
+    def atomic_number(self):
+        if self._atomic_number is not None:
+            return self._atomic_number
+        else:
+            raise InvalidElementError(
+                f"Particle '{self._particle_symbol}' is not an element, "
+                f"so no atomic number is available.")
+
+    @property
+    def mass_number(self):
+        if self._mass_number is not None:
+            return self._mass_number
+        else:
+            raise InvalidIsotopeError(
+                f"Particle '{self._particle_symbol}' is not an isotope, so "
+                f"no mass number is available.")
+
+    @property
+    def lepton_number(self):
+        if self._lepton_number is not None:
+            return self._lepton_number
+        else:
+            raise AtomicError(f"The leption number for "
+                              f"'{self._particle_symbol}' is not available.")
+
+    @property
+    def baryon_number(self):
+        if self._baryon_number is not None:
+            return self._baryon_number
+        elif self._mass_number is not None:
+            return self._mass_number
+        else:
+            raise AtomicError("The baryon number for "
+                              f"'{self._particle_symbol}' is not available.")
+
+    @property
+    def half_life(self):
+        if self._half_life is not None:
+            return self._half_life
+        else:
+            raise MissingAtomicDataError(
+                f"Half-life data for '{self._particle_symbol}' is not "
+                "available.")
+
+    @property
+    def is_stable(self):
+        if self._half_life is not None:
+            return self._half_life == np.inf * u.s
+        else:
+            raise MissingAtomicDataError(
+                f"Stability data for '{self._particle_symbol}' is not "
+                "available.")
+
+    @property
+    def is_antimatter(self):
+        return self._is_antimatter
```

Comment:
I take it this is sort of in preparation for possible future upgrades? Otherwise, you'd just set `self.is_antimatter = self._is_antimatter` in `init`, or even get rid of the latter altogether, right?

Reply:
I'm trying to figure out a way to set it up so that if someone tries to do
```Python
self.is_antimatter = 'maybe?'
```
then they will end up causing an `AttributeError`.   I think we can do that with the [`@property` decorator and setters](http://samwize.com/2012/09/19/how-you-should-write-getter-slash-setter-for-python/).